rocky/katalogus/client.py

```diff
@@ -117,185 +143,248 @@
+class KATalogusClient:
+    def __init__(self, base_uri: str):
+        self.session = httpx.Client(
+            base_url=base_uri,
+            event_hooks={"response": [verify_response]},
+            timeout=settings.ROCKY_OUTGOING_REQUEST_TIMEOUT,
+        )
 
-    def delete_organization(self):
-        response = self.session.delete(self.organization_uri)
-        response.raise_for_status()
+    def health(self) -> ServiceHealth:
+        response = self.session.get("/health")
 
-        logger.info("Deleted organization", organization_code=self.organization)
+        return ServiceHealth.model_validate_json(response.content)
 
-    def get_plugins(self, **params) -> list[Plugin]:
+    def organization_exists(self, organization_code: str) -> bool:
         try:
-            response = self.session.get(f"{self.organization_uri}/plugins", params=params)
-            response.raise_for_status()
-        except httpx.HTTPStatusError as error:
-            raise KATalogusHTTPStatusError(error)
-        return [parse_plugin(plugin) for plugin in response.json()]
+            self.session.get(f"/v1/organisations/{organization_code}")
+        except KATalogusNotAllowedError:
+            return False
 
-    def get_plugin(self, plugin_id: str) -> Plugin:
-        plugin_id = quote(plugin_id)
-        response = self.session.get(f"{self.organization_uri}/plugins/{plugin_id}")
-        response.raise_for_status()
+        return True
 
-        return parse_plugin(response.json())
+    def create_organization(self, organization):
+        self.session.post("/v1/organisations/", json={"id": organization.code, "name": organization.name})
 
-    def get_plugin_schema(self, plugin_id: str) -> dict | None:
-        plugin_id = quote(plugin_id)
-        response = self.session.get(f"{self.organization_uri}/plugins/{plugin_id}/schema.json")
-        response.raise_for_status()
+        logger.info("Created organization", name=organization.name)
 
-        schema = response.json()
-        if not schema:
-            return None
+    def delete_organization(self, organization_code: str):
+        self.session.delete(f"/v1/organisations/{organization_code}")
 
-        try:
-            Draft202012Validator.check_schema(schema)
-        except SchemaError as error:
-            logger.warning("Invalid schema found for plugin %s, %s", plugin_id, error)
-        else:
-            return schema
+        logger.info("Deleted organization", organization_code=organization_code)
 
-        return None
+    def get_plugins(self, organization_code: str, **params) -> list[Plugin]:
+        response = self.session.get(f"/v1/organisations/{organization_code}/plugins", params=params)
+
+        return [parse_plugin(plugin) for plugin in response.json()]
+
+    def get_plugin(self, organization_code: str, plugin_id: str) -> Plugin:
+        response = self.session.get(f"/v1/organisations/{organization_code}/plugins/{quote(plugin_id)}")
```

Comment:
We should quote or sanitize the organization_code in this url.
Better yet, we should provide this function only for an instance of the client bound to a user (and as such can check for the user's permissions for the requested orga), or do we also need this functionality at locations where we dont have a user?

Reply:
I think the KATalogus wrapper serves this purpose, and I (currently) like this service-based design better than forcing backend-choosing-logic behind model specific methods. It would require us to either mock or juggle some global state in testing context (that Django gets away with because they put the necessary effort into managing it). Also: there is nothing preventing me from importing an Organization class, instantiate it with an arbitrary id string, and then access forbidden data through this object, just like there is nothing preventing you from using the "wrong" client and access the wrong data. In essence it's hence a design (pattern) choice, and I think the service-based approach has some advantages of which consistency with other services (e.g. Octopoes repositories, the Bytes implementation, the KATalogus) is not the least.

Perhaps a subsequent issue and discussion topic would be a good idea?